h2o-algos/src/main/java/hex/tree/TreeHandler.java

```diff
@@ -0,0 +1,172 @@
+package hex.tree;
+
+import hex.genmodel.algos.tree.SharedTreeNode;
+import hex.genmodel.algos.tree.SharedTreeSubgraph;
+import hex.schemas.TreeV3;
+import water.MemoryManager;
+import water.api.Handler;
+import water.util.ArrayUtils;
+import water.util.CollectionUtils;
+
+import java.util.*;
+
+/**
+ * Handling requests for various model trees
+ */
+public class TreeHandler extends Handler {
+    private static final int NO_CHILD = -1;
+
+    public TreeV3 getTree(final int version, final TreeV3 args) {
+        validateArgs(args);
+        final SharedTreeModel model = (SharedTreeModel) args.model.key().get();
+        if (model == null) throw new IllegalArgumentException("Unknown tree key: " + args.model.toString());
```

Comment:
`tree_class` should probably be validated as well, you don't want to give out IndexArrayOutOfBoundsException when user makes a mistake

Reply:
Thank you MIchal, fixed.